sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -51,17 +50,28 @@ export class BatchingReceiver extends MessageReceiver {
   }
 
   /**
-   * Clear the token renewal timer and set the `detachedError` property.
-   * @param {AmqpError | Error} [receiverError] The receiver error if any.
+   * To be called when connection is disconnected to gracefully close ongoing receive request.
+   * @param {AmqpError | Error} [connectionError] The connection error if any.
    * @returns {Promise<void>} Promise<void>.
    */
-  async onDetached(receiverError?: AmqpError | Error): Promise<void> {
+  async onDetached(connectionError?: AmqpError | Error): Promise<void> {
     // Clears the token renewal timer. Closes the link and its session if they are open.
     await this._closeLink(this._receiver);
-    this.detachedError = receiverError;
-    if (this.isReceivingMessages && typeof this._finalActionHandler === "function") {
-      this._finalActionHandler();
+
+    if (typeof this._connectionErrorHandler !== "function") {
+      // There is no ongoing receive request, so nothing to do here.
```

Comment:
Should this be `!this.isReceivingMessages`?

Or is there a difference between the two?

Reply:
Yes, but then we had to use `this._connectionErrorHandler!` anyway before using the function